cpp/src/umap/simpl_set_embed/optimize_batch_kernel.cuh

```diff
@@ -198,8 +198,6 @@ CUML_KERNEL void optimize_batch_kernel_reg(T const* head_embedding,
       auto grad_d = T(0.0);
       if (repulsive_grad_coeff > T(0.0))
         grad_d = clip<T>(repulsive_grad_coeff * diff, T(-4.0), T(4.0));
-      else
```

Comment:
This change doesn’t seem to fit within the context of the other changes in this PR. Can you open up another PR for this please so that we can validate and verify this separately from the type changes? 

I’m concerned about changing fundamental math in the solver. Perhaps the reference impl has changed this as well, but I think we can discuss / verify separately.